CHANGES.rst

```diff
@@ -54,6 +54,13 @@
   fields in your subclass before attempting to hash or sort it. See
   `issue 157 <https://github.com/zopefoundation/zope.interface/issues/157>`_.
 
+- Remove unneeded overwrite and call to anyway inherited `__hash__` method
+  from `.declarations.Implements` class. Watching a reindex index process in
+  ZCatalog with on a Py-Spy after 10k samples the time for `.adapter._lookup`
+  was reduced from 27.5s to 18.8s (~1.5x faster). Overall reindex index time
+  shrunk from 369s to 293s (1.26x faster). See
```

Comment:
Is this the same reindex benchmark used for the recent `InterfaceClass.__hash__` changes in #156? I'm just curious because the numbers don't quite make sense taken together and I'm not sure how to reconcile them. Hand-waving at real-world differences, etc, is fine 😄

Reply:
It is the same "benchmark" - a Py-Spy connected to a running exclusively-used and fresh started Plone instance with a copy of the production database. It is the same reindex process (but today on a slighly larger data set than last time). 

The overall  reindex Index performance increment is indeed the same as above. This is just a coincidence (I myself needed to double check too). There is also some derivation possible, this is not a standardized test. With Py-Spy it is now easy to collect data about real world run time timings. 

I can clearly state: A significant reduction is there. For most use cases with less `_lookup` calls it wont be that significant. In this case with heavy `_lookup` calls it is about 1.2x-1.3x faster. 

Thus said, Plone really overuses the ZCA at some places and even at startup or for all Dexterity operations this changes will improve the overall performance (and reduces its carbon footprint, if the servers aren't running on green power).